### DISCUSSION TITLE
Permissions for Inventory Items

### DISCUSSION THREAD

**Discussion by proudbro** [netbox-community/netbox#5733] [Q&A] [answered]
Hi

I have devices which related to tenant "Network Department" (tenant_id = 252, for example) and want to create permission for Additional Components for that devices. 
`dcim.add/change/delete_inventoryitem (Network Department)`
If I just create permission with such constraint `{"tenant__id": 252}` it isn't work which is expected.
Error Output: `Invalid filter for <class 'dcim.models.device_components.InventoryItem'>: Cannot resolve keyword 'tenant' into field. Choices are: _name, asset_tag, child_items, description, device, device_id, discovered, id, label, level, lft, manufacturer, manufacturer_id, name, parent, parent_id, part_id, rght, serial, tagged_items, tags, tree_id`

How to create permission like that employees can add/change/remove additional components for only those devices that belong to a certain tenant (id=252)?

Thanks

**Comment by candlerb** [comment, marked as answer]
Inventory items are not linked directly to tenants, but devices are.  Have you tried `device__tenant` or `device__tenant_id` ?

**Reply by proudbro** [reply]
Thanks a lot. `device__tenant_id` constraint works correctly
